### PR TITLE
Write Tests for Application Utilities | Closes #2242

### LINKED ISSUE #2242: Write Tests for Remaining Application Utilities
- [ ] application/utilities/money
- [ ] application/utilities/text

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Adds tests for `application/utilities/money`
- Closes #2242

### CODE REVIEW

File: assets/prototype/application/utilities/money/test/formatAmount.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { formatAmount } from '../';
+
+describe('formatAmount', () => {
+	it('checks for formatted amount to be a string', () => {
+		const decimalPlaces = 2;
+
+		const format = formatAmount(decimalPlaces);
+		const result = format(5);
+
+		expect(typeof result).toBe('string');
```
Comment (nicolad) [inline assets/prototype/application/utilities/money/test/formatAmount.test.ts]: Should we add here:
`expect(result).toBe('5.00')` ?

File: assets/prototype/application/utilities/money/test/formatAmount.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	});
+
+	it('checks for formatted amount for positive number of deicmal places', () => {
```
Comment (nicolad) [inline assets/prototype/application/utilities/money/test/formatAmount.test.ts]: `decimal places`

File: assets/prototype/application/utilities/money/test/formatAmount.test.ts (hunk 1 above)
Comment (nicolad) [reply assets/prototype/application/utilities/money/test/formatAmount.test.ts]: just read that assertion is `checks for formatted amount to be a string`;
ignore the previous comment

File: assets/prototype/application/utilities/money/test/formatAmount.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	});
+
+	it('checks for formatted amount for 0 deicmal places', () => {
+		const decimalPlaces = 0;
+
+		const format = formatAmount(decimalPlaces);
+
+		expect(format(5.0)).toEqual('5');
+		expect(format(5.2)).toEqual('5');
+		expect(format(-0.5)).toEqual('-1');
```
Comment (nicolad) [inline assets/prototype/application/utilities/money/test/formatAmount.test.ts]: should we test for
`expect(format(5.5)).toEqual(...);`
or
`expect(format(5.49)).toEqual(...);`
?

File: assets/prototype/application/utilities/money/test/amountsMatch.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	it('checks for unequality for different amounts', () => {
+		const amount1 = '8.453';
+		const amount2 = '7.453';
+
+		const result = amountsMatch(amount1, amount2);
+
+		expect(result).toBe(false);
+	});
+});
```
Comment (tn3rb) [inline assets/prototype/application/utilities/money/test/amountsMatch.test.ts]: I think low-level utilities like this could benefit from having more tests written in order to confirm behaviour under any circumstances and edge cases that could potentially occur at runtime (like a var being undefined).
I also think this is a good place to run tests in a loop in order to simplify things, ex:

```js
const testCases = [
	{ 
		desc: 'checks for equality for same amounts as string',
		amount1: '5.623',
		amount2: '5.623',
		result : true,
	},
	{ 
		desc: 'checks for equality for same amounts as string and number',
		amount1: '7.453',
		amount2: 7.453,
		result : true,
	},
	{ 
		desc: 'checks for unequality for different amounts',
		amount1: '8.453',
		amount2: '7.453',
		result : false,
	},
	{ 
		desc: 'returns false when one value is not a number',
		amount1: 'cabbage',
		amount2: '7.453',
		result : false,
	},
	{ 
		desc: 'returns false when one value is null',
		amount1: null,
		amount2: '0',
		result : false,
	},
	// additional tests cases to cover other possibilities that can occur at runtime
];

describe('amountsMatch', () => {
	for (const testCase of testCases) {
		it( testCase.desc, () => {
			const result = amountsMatch( testCase.amount1,  testCase.amount2);
			expect(result).toBe( testCase.result );
		});
	}
}
```

File: assets/prototype/application/utilities/money/test/formatAmount.test.ts (hunk 3 above)
Comment (tn3rb) [reply assets/prototype/application/utilities/money/test/formatAmount.test.ts]: as I wrote above, for low-level utilities, I would prefer to see them getting hammered with as many tests as possible to confirm their behaviour under any and all possible scenarios that could potentially occur at runtime. I also think loops make it easier to write tests for these kinds of functions since the test cases area so straightforward and consistent to test

File: assets/prototype/application/utilities/money/test/parsedAmount.test.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+		const result = parsedAmount('5.256');
+
+		expect(typeof result).toBe('number');
+	});
+
+	it('checks for parsed amount to be same as expected', () => {
+		const result = parsedAmount('8.256');
+
+		expect(result).toEqual(8.256);
+	});
```
Comment (tn3rb) [inline assets/prototype/application/utilities/money/test/parsedAmount.test.ts]: what about `null` or `undefined` or other potential runtime errors?

File: assets/prototype/application/utilities/money/test/useMoneyDisplay.test.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import { ApolloMockedProvider } from '../../../../domain/eventEditor/context/test';
+
+describe('useMoneyDisplay', () => {
+	const wrapper = ApolloMockedProvider();
+
+	it('checks for existence of properties', () => {
+		const { result } = renderHook(() => useMoneyDisplay(), { wrapper });
+
+		expect(result.current).toHaveProperty('beforeAmount');
+		expect(result.current).toHaveProperty('afterAmount');
```
Comment (tn3rb) [inline assets/prototype/application/utilities/money/test/useMoneyDisplay.test.ts]: can we also test the values for these properties based on the mock JS data?

Comment (manzoorwanijk) [reply assets/prototype/application/utilities/money/test/useMoneyDisplay.test.ts]: Those are calculated, which mean those calculation will need to be pulled/duplicated into tests

File: assets/prototype/application/utilities/money/test/amountsMatch.test.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+		amount2: '0',
+		result: false,
+	},
+	{
+		desc: 'returns true when both are string 0',
+		amount1: '0',
+		amount2: '0',
+		result: true,
+	},
+];
```
Comment (nicolad) [inline assets/prototype/application/utilities/money/test/amountsMatch.test.ts]: Noooooice :+1:
